parts/k8s/kubernetesmasterresources.t

```diff
@@ -866,14 +866,15 @@
        },
        "apiVersion": "[variables('apiVersionCompute')]",
        "location": "[resourceGroup().location]",
-       {{if (not UserAssignedIDEnabled)}}
+       {{if UserAssignedIDEnabled}}
        "dependsOn": [
          "[concat('Microsoft.Compute/virtualMachines/', variables('masterVMNamePrefix'), copyIndex())]",
-         "[concat('Microsoft.Authorization/roleAssignments/', guid(concat('Microsoft.Compute/virtualMachines/', variables('masterVMNamePrefix'), copyIndex(), 'vmidentity')))]"
+         "[concat('Microsoft.Authorization/roleAssignments/', guid(concat(variables('userAssignedID'), 'roleAssignment')))]"
```

Comment:
Nit: Can we confine the  {{if UserAssignedIDEnabled}} and {{else}} blocks to span only Lines 872 and 877? That will remove some redundancy

Reply:
Yep, makes sense, Will add it in the upcoming iteration.